src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -346,8 +346,12 @@ protected final void runLeaf(Statement statement, Description description,
 
     @Override
     public Description getDescription() {
-        Description description = Description.createSuiteDescription(getName(),
-                getRunnerAnnotations());
+        Description description =
+                Description.createSuiteDescription(
+                        getTestClass().getJavaClass(),
```

Comment:
It looks like `getJavaClass()` can in theory return `null`. Just to be safe, I suggest you check for that here, and if it is `null` use `getName()` like we did before.